mail_restrict_auto_follower/models/mail_thread.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, api
+from openerp.tools.safe_eval import safe_eval
+
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    @api.model
+    def _mail_restrict_follower_get_domain(self):
+        parameter_name = 'mail_restrict_auto_follower.domain'
+        return self.env['ir.config_parameter'].get_param(
+            '%s.%s' % (parameter_name, self._name),
+            default=self.env['ir.config_parameter'].get_param(
+                parameter_name, default='[]'))
+
+    @api.multi
+    def message_subscribe(self, partner_ids, subtype_ids=None):
+        if self.env.context.get('allow_follower_subscription'):
```

Comment:
I think you should override https://github.com/OCA/OCB/blob/8.0/addons/mail/mail_thread.py#L1874 to set a context key there, and react to it here. This we, the default behavior is what standard does, and only when called by `message_auto_subscribe`, we do the filtering. This is safer.